includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1656,34 +1663,46 @@ private function get_ancestor_with_matching_spec_name( $node, $ancestor_spec_nam
 	 * Also adds them to the stack for processing by the sanitize() function.
 	 *
 	 * @since 0.3.3
+	 * @since 1.0 Fix silently removing unrecognized elements.
+	 * @see https://github.com/Automattic/amp-wp/issues/1100
 	 *
 	 * @param DOMNode $node Node.
 	 */
 	private function replace_node_with_children( $node ) {
 
+		// If node has no children or no parent, just remove the node.
 		if ( ! $node->hasChildNodes() || ! $node->parentNode ) {
```

Comment:
Should not there be a `$this->should_sanitize_validation_error( array(), compact( 'node' ) )` call in here as well? Otherwise the removal won't be reported as a validation error.

Reply:
@westonruter the `if` code block flows through to `remove_node()` where it’s handled.  I did confirm it there.  When I write the error unit tests, I’ll validate that too to ensure we have reporting for both paths.